ui/components/app/toast-master/toast-master.js

```diff
@@ -0,0 +1,299 @@
+/* eslint-disable react/prop-types -- TODO: upgrade to TypeScript */
+
+import React, { useState } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { useHistory, useLocation } from 'react-router-dom';
+import { MILLISECOND, SECOND } from '../../../../shared/constants/time';
+import {
+  PRIVACY_POLICY_LINK,
+  SURVEY_LINK,
+} from '../../../../shared/lib/ui-utils';
+import {
+  BorderColor,
+  BorderRadius,
+  IconColor,
+  TextVariant,
+} from '../../../helpers/constants/design-system';
+import {
+  DEFAULT_ROUTE,
+  REVIEW_PERMISSIONS,
+} from '../../../helpers/constants/routes';
+import { getURLHost } from '../../../helpers/utils/util';
+import { useI18nContext } from '../../../hooks/useI18nContext';
+import { usePrevious } from '../../../hooks/usePrevious';
+import {
+  getCurrentNetwork,
+  getOriginOfCurrentTab,
+  getSelectedAccount,
+  getSwitchedNetworkDetails,
+  getUseNftDetection,
+} from '../../../selectors';
+import {
+  addPermittedAccount,
+  clearSwitchedNetworkDetails,
+  hidePermittedNetworkToast,
+} from '../../../store/actions';
+import {
+  AvatarAccount,
+  AvatarAccountSize,
+  AvatarNetwork,
+  Icon,
+  IconName,
+} from '../../component-library';
+import { Toast, ToastContainer } from '../../multichain';
+import { SurveyToast } from '../../ui/survey-toast';
+import {
+  selectNftDetectionEnablementToast,
+  selectShowConnectAccountToast,
+  selectShowPrivacyPolicyToast,
+  selectShowSurveyToast,
+  selectSwitchedNetworkNeverShowMessage,
+} from './selectors';
+import {
+  setNewPrivacyPolicyToastClickedOrClosed,
+  setNewPrivacyPolicyToastShownDate,
+  setShowNftDetectionEnablementToast,
+  setSurveyLinkLastClickedOrClosed,
+  setSwitchedNetworkNeverShowMessage,
+} from './utils';
+
+export function ToastMaster() {
+  const location = useLocation();
+
+  const onHomeScreen = location.pathname === DEFAULT_ROUTE;
+
+  return (
+    onHomeScreen && (
+      <ToastContainer>
+        <SurveyToast />
+        <ConnectAccountToast />
+        <SurveyToastMayDelete />
+        <PrivacyPolicyToast />
+        <SwitchedNetworkToast />
+        <NftEnablementToast />
+        <PermittedNetworkToast />
+      </ToastContainer>
+    )
+  );
+}
+
+function ConnectAccountToast() {
+  const t = useI18nContext();
+  const dispatch = useDispatch();
+
+  const [hideConnectAccountToast, setHideConnectAccountToast] = useState(false);
+  const account = useSelector(getSelectedAccount);
+
+  // If the account has changed, allow the connect account toast again
+  const prevAccountAddress = usePrevious(account?.address);
+  if (account?.address !== prevAccountAddress && hideConnectAccountToast) {
+    setHideConnectAccountToast(false);
```

Comment:
This should be part of a useEffect instead of running it from the root of the component